src/FileDownloader.ts

```diff
@@ -84,16 +85,23 @@ export default class FileDownloader implements IFileDownloader {
                 onDownloadProgressChange
             );
 
-            const writeStream: Writable = settings?.shouldUnzip ?? false
-                ? ZipDecompressor(tempFileDownloadPath)
-                : fs.createWriteStream(tempFileDownloadPath);
-
-            // Start the download and wait for it to completely finish writing to disk
-            // The first element in the pipeline must be the readableStream and the last element must be the writableStream
+            const writeStream = fs.createWriteStream(tempZipFileDownloadPath);
             const pipelinePromise = pipelineAsync([downloadStream, writeStream]);
             const writeStreamClosePromise = new Promise(resolve => writeStream.on(`close`, resolve));
             await Promise.all([pipelinePromise, writeStreamClosePromise]);
 
+            const unzipDownloadedFileAsyncFn = async (): Promise<void> => {
```

Comment:
Nit: it looks strange to even be able to enter this function if `settings?.shouldUnzip` isn't true.

Reply:
I would update the variable name.